### PR TITLE
fix(IconButton): dynamic size prop

### LINKED ISSUE #10293: Add `size` prop to `IconButton` and correctly forward it to `Button`
As referenced in: https://github.com/carbon-design-system/carbon/discussions/10247#discussioncomment-1798312 the `IconButton` needs to correctly set a `size` prop and forward it along to `Button`

### PR DESCRIPTION
**Pull request by sstrubberg**
closes #10293 

It was hard-coded to "sm". Now it accepts a prop. Defaults to 'md'.

### CODE REVIEW

**Review by joshblack — APPROVED** [review]
Looks great!

**Comment by joshblack** [comment]
@laurenmrice just wanted to double check, what sizes are allowed for `IconButton`? 👀

**Comment by laurenmrice** [comment]
@joshblack The Icon button can be Large (48px), Medium (40px), or Small (32px).

File: packages/react/src/components/IconButton/index.js [hunk 1]
```diff
@@ -86,6 +87,11 @@ IconButton.propTypes = {
    * Specify the duration in milliseconds to delay before hiding the tooltip
    */
   leaveDelayMs: PropTypes.number,
+
+  /**
+   * Specify the size of the Button. Defaults to `md`.
+   */
+  size: PropTypes.oneOf(['xs', 'sm', 'md', 'lg', 'xl', '2xl']),
```
Comment (joshblack) [inline packages/react/src/components/IconButton/index.js]:
```suggestion
  size: PropTypes.oneOf(['sm', 'md', 'lg']),
```

**Review by tw15egan — APPROVED** [review]
LGTM! 👍🏻 ✅

**Pull request merged**